File: server/src/main/java/io/github/titaniumcoder/toggl/reporting/config/TogglConfiguration.java
package io.github.titaniumcoder.toggl.reporting.config;

import io.micronaut.context.annotation.ConfigurationProperties;
import lombok.Data;

@ConfigurationProperties("application")
@Data
public class TogglConfiguration {
    private String apiToken = "";
    private Long workspaceId = -1L;
    private SecurityEncodingConfiguration security = new SecurityEncodingConfiguration();

    @Data
    @ConfigurationProperties("security")
    public static class SecurityEncodingConfiguration {
        private String secret = "";
        private String username = "";
        private String password = "";
        private Long expiration = -1L;
    }
}

File: server/src/main/kotlin/io/github/titaniumcoder/toggl/reporting/config/TogglConfiguration.kt
package io.github.titaniumcoder.toggl.reporting.config

import org.springframework.boot.context.properties.ConfigurationProperties
import javax.validation.constraints.NotBlank

@ConfigurationProperties(prefix = "application")
class TogglConfiguration {
    @NotBlank
    lateinit var apiToken: String

    @NotBlank
    var workspaceId: Long = -1L

    var security: SecurityEncodingConfiguration = SecurityEncodingConfiguration()

    class SecurityEncodingConfiguration {
        @NotBlank
        lateinit var username: String

        @NotBlank
        lateinit var password: String
    }
}

